src/NativeScript/JSErrors.mm

```diff
@@ -49,12 +96,23 @@ void reportErrorIfAny(JSC::ExecState* execState, JSC::CatchScope& scope) {
             id discardExceptionsValue = [[TNSRuntime current] appPackageJson][@"discardUncaughtJsExceptions"];
             bool discardExceptions = [discardExceptionsValue boolValue];
             scope.clearException();
-            reportFatalErrorBeforeShutdown(execState, exception, true, discardExceptions);
+            if (discardExceptions) {
+                reportDiscardedError(execState, exception);
+            } else {
+                reportFatalErrorBeforeShutdown(execState, exception, true);
+            }
         }
     }
 }
 
-void reportFatalErrorBeforeShutdown(ExecState* execState, Exception* exception, bool callUncaughtErrorCallbacks, bool dontCrash) {
+void reportDiscardedError(ExecState* execState, Exception* exception) {
+    NakedPtr<Exception> errorCallbackException;
+    GlobalObject* globalObject = static_cast<GlobalObject*>(execState->lexicalGlobalObject());
+    globalObject->callJsDiscardedErrorCallback(execState, exception, errorCallbackException);
```

Comment:
We should check whether `errorCallbackException` is set and log it. This way if the handler throws an error for some reason users will be notified. Right now they cannot detect it. We should do the same in `reportFatalErrorBeforeShutdown` as well.